lib/node_modules/@stdlib/array/complex64/test/test.index_of.js

```diff
@@ -0,0 +1,205 @@
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a complex number', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.indexOf( value );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a second argument which is not a nonnegative integer', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		-5,
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.indexOf( new Complex64( 1.0, 1.0 ), value );
+		};
+	}
+});
+
+tape( 'the method returns `-1` if provided an empty complex number array', function test( t ) {
+	var arr;
+	var idx;
+
+	arr = new Complex64Array();
+	idx = arr.indexOf( new Complex64( 1.0, 1.0 ) );
+
+	t.strictEqual( idx, -1, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the method returns `-1` if a complex number is not found', function test( t ) {
+	var idx;
+	var arr;
+
+	arr = new Complex64Array( 10 );
+	idx = arr.indexOf( new Complex64( 1.0, 1.0 ) );
+
+	t.strictEqual( idx, -1, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the method returns the index of the first match if a complex number is found', function test( t ) {
+	var idx;
+	var arr;
+
+	arr = new Complex64Array( 10 );
+	arr.set( [ 1.0, 1.0 ], 0 );
+	arr.set( [ 2.0, 2.0 ], 1 );
+	arr.set( [ 3.0, 3.0 ], 2 );
+	arr.set( [ 2.0, 2.0 ], 3 );
+	idx = arr.indexOf( new Complex64( 2.0, 2.0 ) );
+
+	t.strictEqual( idx, 1, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the method returns `-1` if provided fromIndex exceeds complex number array length', function test( t ) {
+	var idx;
+	var arr;
+
+	arr = new Complex64Array( 10 );
+	idx = arr.indexOf( new Complex64( 1.0, 1.0), 20 );
+
+	t.strictEqual( idx, -1, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the method supports specifying `fromIndex`', function test( t ) {
+	var idx;
+	var arr;
+
+	arr = new Complex64Array( 10 );
+	arr.set( [ 1.0, 1.0 ], 0 );
+	arr.set( [ 2.0, 2.0 ], 1 );
+	arr.set( [ 3.0, 3.0 ], 2 );
+	arr.set( [ 2.0, 2.0 ], 3 );
+
+	idx = arr.indexOf( new Complex64( 2.0, 2.0 ), 0 );
+	t.strictEqual( idx, 1, 'returns expected value' );
+
+	idx = arr.indexOf( new Complex64( 2.0, 2.0 ), 2 );
+	t.strictEqual( idx, 3, 'returns expected value' );
+
+	idx = arr.indexOf( new Complex64( 2.0, 2.0 ), 5 );
```

Comment:
```suggestion
	idx = arr.indexOf( new Complex64( 3.0, 3.0 ), 3 );
```

This is similar to the test above for when `fromIndex` exceeds the array length. Changing to not find a match when providing a `fromIndex` argument.